app.ts

```diff
@@ -0,0 +1,366 @@
+
+let state = {
+	'records': Math.floor((window.innerHeight - 160) / 40), // Estimate of available table space
+	'trimStart': 0,
+	'trimEnd': 10,
+	'countRec': 0
+}
+
+
+// These are 'technically' constants
+let RECORDCOUNT = 350;
+let HEADERS = ["ID", "City", "Population"];
+
+// 'Global' get element 
+function el(element: string) {
+	return document.getElementById(element);
+}
+
+
+// Fetch headers and record count
+async function getData() {
+	// API calls for record count and headers
+	RECORDCOUNT = await (await fetch('/recordCount')).json();
+	HEADERS = await (await fetch('/columns')).json();
+
+	// Populate table with fetched data
+	loadIntoTable(el('content-table'));
+}
+
+
+// Search entered ID
+async function searchFunction() {
+	let idString = $('#id-search').val()
+	let id = Number($('#id-search').val());
+	let pageInfo = el('page-number');
+	let numRecords = RECORDCOUNT - 1;
+
+
+	if (id < 0 || id > numRecords || Number.isNaN(id) || idString == "") {
+		// User info: Display error message
+		pageInfo!.innerHTML = `<p>No records to display</p>`;
+	}
+	else {
+		// Use entered ID to calculate what page it is on
+		if ((RECORDCOUNT - 1) - id >= state.records) {
+			state.trimStart = id;
+			state.trimEnd = state.trimStart + (state.records - 1);
+		}
+		else {
+			state.trimEnd = RECORDCOUNT - 1;
+			state.trimStart = state.trimEnd - state.records + 1;
+		}
+		loadIntoTable(el("table"));
+	}
+}
+
+
+// Add rows to table
+async function addRows(link: string) {
```

Comment:
You can change the **addRows** function to accept a boolean as a parameter e.g. **prepend**. This boolean can determine if you want to append or prepend. Then it won't be necessary for the prependRows function. They have the exact same functionality.

```suggestion
async function addRows(link: string, prepend?: boolean) {
```